public/app/features/dashboard/components/DashLinks/DashLinksEditorCtrl.ts

```diff
@@ -65,6 +65,7 @@ export class DashLinksEditorCtrl {
   }
 
   saveLink() {
+    this.dashboard.links = [...this.dashboard.links];
```

Comment:
I think this needs to do a deep clone as the DashboardLinksDashboard checks link equality to figure out out if another search for dashboards should be triggered.

You could test this by changing the tag in a link definition and then go back to dashboard, does list of links change?

Reply:
You are right, fixed